url1cbaffb197_foxking0416_AlgorithmTraining_tgz-pJ8-algorithm_leetcode_MainLeetJ8/src/datastructure/binarytree/BinaryTree.java

```diff
@@ -0,0 +1,1106 @@
+		if(rootNode == null){
+			this.rootNode = newNode;
+		}
+		else{
+			insertNode(this.rootNode, newNode);
+		}
+		count++;
+	}
+	
+	protected void insertNode(TreeNode currentNode, TreeNode newNode){
+		if(newNode.val < currentNode.val){
+			if(currentNode.getLeftLeafNode() == null){
+				currentNode.setLeftLeafNode(newNode);
+			}
+			else {
+				insertNode(currentNode.getLeftLeafNode(), newNode);
+			}
+		}
+		else{
+			if(currentNode.getRightLeafNode() == null){
+				currentNode.setRightLeafNode(newNode);
+			}
+			else {
+				insertNode(currentNode.getRightLeafNode(), newNode);
+			}
+		}
+	}
+	
+	public TreeNode getRootNode(){
+		return this.rootNode;
+	}
+	
+	public boolean contains(TreeNode currentNode, TreeNode newNode){
+		
+		if(currentNode == null)
+			return false;
+		
+		if(currentNode.val == newNode.val){
+			return true;
+		}
+		else{
+			if(newNode.val < currentNode.val){
+				return contains(currentNode.getLeftLeafNode(), newNode);
+			}
+			else {
+				return contains(currentNode.getRightLeafNode(), newNode);
+			}
+		}
+	}
+	
+	public TreeNode findNode(TreeNode currentNode, TreeNode newNode){
+		if(currentNode == null)
+			return null;
+		
+		if(currentNode.val == newNode.val){
+			return currentNode;
+		}
+		else{
+			if(newNode.val < currentNode.val){
+				return findNode(currentNode.getLeftLeafNode(), newNode);
+			}
+			else {
+				return findNode(currentNode.getRightLeafNode(), newNode);
+			}
+		}
+	}
+	
+
+	public int findMin(TreeNode currentNode){
+		if(currentNode == null)
+			return -1;
+		else{
+			if(currentNode.getLeftLeafNode() == null)
+				return currentNode.val;
+			else {
+				return findMin(currentNode.getLeftLeafNode());
+			}
+		}
+	}
+	
+	public int findMax(TreeNode currentNode){
+		if(currentNode == null)
+			return -1;
+		else {
+			if(currentNode.getRightLeafNode() == null)
+				return currentNode.val;
+			else {
+				return findMax(currentNode.getRightLeafNode());
+			}
+		}
+	}
+	
+	public int height(TreeNode node){
+		
+		if(node == null)
+			return -1;
+		
+		int leftHeight = height(node.getLeftLeafNode());
+		int rightHeight = height(node.getRightLeafNode());
+		
+		if(leftHeight > rightHeight)
+			return leftHeight + 1;
+		else 
+			return rightHeight + 1;
+		
+		
+	}
+	
+	public int minDepth(TreeNode rNode) {
+        if(rNode == null)
+        	return 0;
+        
+        if(rNode.getLeftLeafNode() == null && rNode.getRightLeafNode() == null)
+        	return 1;
+        else if(rNode.getLeftLeafNode() == null)
+        	return minDepth(rNode.getRightLeafNode()) + 1;
+        else if(rNode.getRightLeafNode() == null)
+        	return minDepth(rNode.getLeftLeafNode()) + 1;
+        
+        int leftHeight = minDepth(rNode.getLeftLeafNode());
+        int rightHeight = minDepth(rNode.getRightLeafNode());
+        
+        if(leftHeight < rightHeight)
+        	return leftHeight + 1;
+        else 
+			return rightHeight + 1;
+		
+    }
+	
+	public void remove(TreeNode removeNode){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.